src/ops/init/mod.rs

```diff
@@ -0,0 +1,37 @@
+//! Bootstrap a new lorri project
+
+use crate::ops::{ok, ExitError, OpResult};
+use std::fs::File;
+use std::io;
+use std::io::Write;
+use std::path::Path;
+
+fn create_if_missing(path: &Path, contents: &str) -> Result<(), io::Error> {
+    if path.exists() {
+        println!("- Skipping existing {}", path.display());
+        Ok(())
+    } else {
+        let mut f = File::create(path)?;
+        f.write_all(contents.as_bytes())?;
+        println!("- Writing {}", path.display());
+        Ok(())
+    }
+}
+
+fn to_op(e: Result<(), io::Error>) -> OpResult {
+    match e {
+        Ok(_) => ok(),
+        Err(e) => ExitError::errmsg(format!("{}", e)),
+    }
+}
+
+/// See the documentation for lorri::cli::Command::Init for
+/// more details
+pub fn main(default_shell: &str, default_envrc: &str) -> OpResult {
+    println!("Bootstrapping a new lorri project\n");
+
+    to_op(create_if_missing(Path::new("./shell.nix"), default_shell))?;
+    to_op(create_if_missing(Path::new("./.envrc"), default_envrc))?;
```

Comment:
I’d add the better “exists” messages I listed in #41 here:

* “shell.nix exists, skipping. Make sure it is of a form that works with nix-shell.”
* “.envrc exists, skipping. Please add eval "$(lorri direnv)" to it to set up lorri support.”

Reply:
A bit wordy for my taste but on the other hand i don't have strong feelings about that ;-)